### PR TITLE
Modal component

### LINKED ISSUE #70: Add a modal overlay component in React
### Feature request-
Add a modal overlay component in React.
Keep the code super simple so that it is reusable and use React portals (if necessary) to make sure the modal comes over rest of the elements in the DOM.

LGM-SOC

<hr/>

**Note:**

-   If you want to work on an issue, you should check if it has already been assigned to anyone. **If the issue is free** you can comment `/assign` to get the issue assigned to you.
-   If you are raising a new issue and want to work on it then also you should comment `/assign` under the issue to get it auto assigned.
- It is adviced to take up/ work on only 2 issues at the same time.
-   Please **refrain from** adding labels to your issue/pull-request on your own. It is the job of the Project Admin and the Mentors to review your issue/pull-request and add labels accordingly.

### PR DESCRIPTION
**Pull request by okarin1511**
## What is the change?
Added a Modal component with attributes like header, heading, body, button.

## Related issue?
closes: #70 

## How was it tested?
Tested it on the local development server.

## Checklist:
Before you create this PR, confirm all the requirements listed below by checking the checkboxes `[x]`:

-   [x] Have you followed the [Contribution Guidelines](https://github.com/ALPHAVIO/BlogSite/blob/master/CONTRIBUTING.md) while contributing.
-   [x] Have you checked there aren't other open [Pull Requests](https://github.com/ALPHAVIO/Mirage-UI/pulls) for the same update/change?
-   [x] Have you made corresponding changes to the documentation?
-   [x] Have you tested the code before submission?
-   [x] Did you lint and format your code before making the Pull Request?
-   [x] Have you made the GIF for the component you added?

## GIF you created:
I inserted an image instead.

image

### CODE REVIEW

File: React/Modal/modal-component/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
Comment (Pranav016) [inline React/Modal/modal-component/README.md]: Remove this file

File: React/Modal/modal-component/public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
```
Comment (Pranav016) [inline React/Modal/modal-component/public/index.html]: Remove unnecessary code and comments

File: React/Modal/modal-component/public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (Pranav016) [inline React/Modal/modal-component/public/index.html]: Remove favicon and logo links from this

File: React/Modal/modal-component/public/manifest.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (Pranav016) [inline React/Modal/modal-component/public/manifest.json]: remove this file

File: React/Modal/modal-component/public/robots.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
```
Comment (Pranav016) [inline React/Modal/modal-component/public/robots.txt]: remove file

File: React/Modal/modal-component/src/App.css [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+  color: white;
+}
+
+.App-link {
+  color: #61dafb;
+}
+
+@keyframes App-logo-spin {
```
Comment (Pranav016) [inline React/Modal/modal-component/src/App.css]: This css is not even used in the actual modal, please check if this is not necessary then remove the file

File: React/Modal/modal-component/src/App.test.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
```
Comment (Pranav016) [inline React/Modal/modal-component/src/App.test.js]: remove file

File: React/Modal/modal-component/src/logo.svg [hunk 8]
```diff
@@ -0,0 +1 @@
...21.2 430c-9.3-9.6-18.6-20.3-27.8-32 9 .4 18.2.7 27.5.7 9.4 0 18.7-.2 27.8-.7-9 11.7-18.3 22.4-27.5 32zm-74.4-58.9c-14.2-2.1-27.9-4.7-41-7.9 3.7-12.9 8.3-26.2 13.5-39.5 4.1 8 8.4 16 13.1 24 4.7 8 9.5 15.8 14.4 23.4zM420.7 163c9.3 9.6 18.6 20.3 27.8 32-9-.4-18.2-.7-27.5-.7-9.4 0-18.7.2-27.8.7 9-11.7 18.3-22.4 27.5-32zm-74 58.9c-4.9 7.7-9.8 15.6-14.4 23.7-4.6 8-8.9 16-13 24-5.4-13.4-10-26.8-13.8-39.8 13.1-3.1 26.9-5.8 41.2-7.9zm-90.5 125.2c-35.4-15.1-58.3-34.9-58.3-50.6 0-15.7 22.9-35.6 58.3-50.6 8.6-3.7 18-7 27.7-10.1 5.7 19.6 13.2 40 22.5 60.9-9.2 20.8-16.6 41.1-22.2 60.6-9.9-3.1-19.3-6.5-28-10.2zM310 490c-13.6-7.8-19.5-37.5-14.9-75.7 1.1-9.4 2.9-19.3 5.1-29.4 19.6 4.8 41 8.5 63.5 10.9 13.5 18.5 27.5 35.3 41.6 50-32.6 30.3-63.2 46.9-84 46.9-4.5-.1-8.3-1-11.3-2.7zm237.2-76.2c4.7 38.2-1.1 67.9-14.6 75.8-3 1.8-6.9 2.6-11.5 2.6-20.7 0-51.4-16.5-84-46.6 14-14.7 28-31.4 41.3-49.9 22.6-2.4 44-6.1 63.6-11 2.3 10.1 4.1 19.8 5.2 29.1zm38.5-66.7c-8.6 3.7-18 7-27.7 10.1-5.7-19.6-13.2-40-22.5-60.9 9.2-20.8 16.6-41.1 22.2-60.6 9.9 3.1 19.3 6.5 28.1 10.2 35.4 15.1 58.3 34.9 58.3 50.6-.1 15.7-23 35.6-58.4 50.6zM320.8 78.4z"/><circle cx="420.9" cy="296.5" r="45.7"/><path d="M520.5 78.1z"/></g></svg>
```
Comment (Pranav016) [inline React/Modal/modal-component/src/logo.svg]: remove file

File: React/Modal/modal-component/src/setupTests.js [hunk 9]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
```
Comment (Pranav016) [inline React/Modal/modal-component/src/setupTests.js]: remove

File: React/Modal/modal-component/src/reportWebVitals.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+const reportWebVitals = onPerfEntry => {
```
Comment (Pranav016) [inline React/Modal/modal-component/src/reportWebVitals.js]: remove

File: React/README.md [hunk 11]
```diff
@@ -1 +1,26 @@
 # React components
+
+## Modal Component:
```
Comment (Pranav016) [inline React/README.md]: This README should be inside the component, it acts as a documentation for the component. You have put it in the wrong place. Also use the README template in the repository to make a README for your component.

**Review by Pranav016 — CHANGES_REQUESTED** [review]
Pull the recent changes and see how I have added a new react component, follow the same structure. You can take reference on how to cleanup the folder after using create-react-app. Also you can take reference for the README of the component.
- Put your modal component inside a new folder `Modals`
- Cleanup the repo, remove unwanted/ unused images and favicons and files. They bloat the project.
- Add a demo video or GIF, image doesn't help to see functioning of the component.
- Resolve merge conflicts

**Comment by Pranav016** [comment]
Post your video/ GIF here in the comments first before pushing your new changes.

**Comment by okarin1511** [comment]
I've made the changes, please review it.

Please let me know if you want me to change anything else

**Review by Pranav016 — APPROVED** [review]

**Pull request merged**